### PR TITLE
CU-27gyq6e Login Page

### PR DESCRIPTION
**Pull request by cristhianescobar**
Implemented login page, and added a "Continue as Guest" button since we don't have account creation setup just yet.

TODO: Introduce Login UI Test 

loginscreen

### CODE REVIEW

**Comment by newmskywalker** [comment]
Task linked: [CU-27gyq6e Android: Develop the login page UI](https://app.clickup.com/t/27gyq6e)

File: android/app-project-newm/src/main/java/io/projectnewm/screens/login/LoginScreen.kt [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+import io.projectnewm.screens.login.email.EmailState
+import io.projectnewm.screens.login.password.Password
+import io.projectnewm.screens.login.password.PasswordState
+import io.projectnewm.utilities.ToBeImplemented
+import io.projectnewm.utilities.shortToast
+
+@Composable
+fun LoginScreen(
+    onSignInSubmitted: (email: String, password: String) -> Unit,
+    scrollableState: ScrollState = rememberScrollState()
```
Comment (andymiller813) [inline android/app-project-newm/src/main/java/io/projectnewm/screens/login/LoginScreen.kt]: nit: 
```suggestion
    scrollState: ScrollState = rememberScrollState()
```

File: android/app-project-newm/src/main/java/io/projectnewm/screens/login/LoginScreen.kt [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            .padding(horizontal = 30.dp)
+            .verticalScroll(scrollableState),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center
+    ) {
+
+        Spacer(modifier = Modifier.height(50.dp))
+
+        LoginPageMainImage(R.drawable.ic_newm_logo)
+        LoginPageMainTextImage(textImage = R.drawable.login_enter_newminiverse)
```
Comment (andymiller813) [inline android/app-project-newm/src/main/java/io/projectnewm/screens/login/LoginScreen.kt]: It's not letting me comment on the icon file so I'll comment here:
Let's prefix icons with ic_ and backgrounds with bg_

Also typo `newminiverse` -> `newmiverse`

File: android/app-project-newm/src/main/java/io/projectnewm/screens/login/email/EmailState.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package io.projectnewm.screens.login.email
+
+import io.projectnewm.screens.login.TextFieldState
+import java.util.regex.Pattern
+
+// Consider an email valid if there's some text before and after a "@"
+private const val EMAIL_VALIDATION_REGEX = "^(.+)@(.+)\$"
+
+class EmailState : TextFieldState(validator = ::isEmailValid, errorFor = ::emailValidationError)
```
Comment (andymiller813) [inline android/app-project-newm/src/main/java/io/projectnewm/screens/login/email/EmailState.kt:7]: Marty added a LoginFieldValidator in commonMain/login

File: android/app-project-newm/src/main/java/io/projectnewm/screens/login/password/PasswordState.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            null
+        }
+    }
+}
+
+private fun passwordAndConfirmationValid(password: String, confirmedPassword: String): Boolean {
+    return isPasswordValid(password) && password == confirmedPassword
+}
+
+private fun isPasswordValid(password: String): Boolean {
```
Comment (andymiller813) [inline android/app-project-newm/src/main/java/io/projectnewm/screens/login/password/PasswordState.kt:25]: Use shared password validator.  Right now it just checks if the field is blank so you could add these checks there as well

File: buildSrc/src/main/java/Google.kt [hunk 5]
```diff
@@ -21,6 +21,8 @@ object Google {
     const val navigationUiKtx = "androidx.navigation:navigation-ui-ktx:${Versions.androidxNavigation}"
     const val splashScreen = "androidx.core:core-splashscreen:${Versions.androidxSplashScreen}"
 
+    const val materialIconsExtended = "androidx.compose.material:material-icons-extended:1.2.0-alpha04"
```
Comment (andymiller813) [inline buildSrc/src/main/java/Google.kt]: Add version to versions file

File: android/app-project-newm/src/main/java/io/projectnewm/screens/login/LoginScreen.kt [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        contentScale = ContentScale.FillHeight,
+    )
+}
+
+@Composable
+private fun LoginPageBackgroundImage(@DrawableRes backgroundImage: Int) {
+    Image(
+        modifier = Modifier
+            .fillMaxWidth()
+            .fillMaxWidth(),
```
Comment (andymiller813) [inline android/app-project-newm/src/main/java/io/projectnewm/screens/login/LoginScreen.kt:194]: Think one of these was supposed to be height?

**Review by andymiller813 — APPROVED** [review]

**Pull request merged**